cli/src/cmd/cast/wallet.rs

```diff
@@ -171,8 +176,24 @@ impl WalletSubcommands {
                     hex::encode(wallet.signer().to_bytes()),
                 );
             }
-            WalletSubcommands::Address { wallet } => {
+            WalletSubcommands::Address { wallet, private_key_override } => {
                 // TODO: Figure out better way to get wallet only.
+                let wallet = if Option::is_some(&private_key_override) {
```

Comment:
It looks like all fields of "Wallet" are the default except private_key?

then we can make this nicer by

1. add `Default` to derives of `Wallet`:

 ```suggestion
   let wallet = private_key_override.map(|private_key| Wallet {private_key, ..Default::default()}).unwrap_or(wallet);
```